tests/zero-downtime-scaling/zero-downtime-scaling.md

```diff
@@ -0,0 +1,381 @@
+  either the client or the backend. This means that unless all those connections are closed by clients/backends before
+  or during an upgrade (which is highly unlikely), NGINX will not terminate, which means Kubernetes will kill NGINX. As
+  a result, the clients will see the connections abruptly closed and thus experience downtime.
+
+  As a result, we *will not* use any long-live connections in this test, because NGF cannot support zero-downtime
+  upgrades in this case.
+
+- Scaling to over 25 replicas.
+
+## Test Environment
+
+The scaling tests will be performed on two different Kubernetes clusters:
+
+- A Kubernetes cluster with 10 nodes on GKE:
+  - Node: n2d-standard-4 (4 vCPU, 8GB memory).
+  - Enabled GKE logging.
+- A Kubernetes cluster with 25 nodes on GKE:
+  - Node: n2d-standard-4 (4 vCPU, 8GB memory).
+  - Enabled GKE logging.
+
+The 10 Node cluster will cover the case where multiple NGF Pods are running on the same Node, whereas the 25 Node
+cluster will test 1 NGF Pod per Node.
+
+The rest of the test environment is the same for both cases:
+
+- Tester VMs:
+  - Configuration:
+    - Debian
+    - Install packages: wrk, curl, gnuplot.
+  - Location - same zone as the Kubernetes cluster.
+  - First VM - for HTTP traffic.
+  - Second VM - for sending HTTPs traffic.
+- NGF
+  - Deployment with 1 replicas (during the test we will scale this up to a max of 25 replicas).
+  - Exposed via a Service with type LoadBalancer, private IP.
+  - Configure delayed termination with a 40-second sleep.
+  - Gateway, two listeners - HTTP and HTTPs.
+  - Two apps:
+    - Coffee - 10 replicas.
+    - Tea - 10 replicas.
+  - Two HTTPRoutes
+    - Coffee (HTTP).
+    - Tea (HTTPS).
+
+Notes:
+
+- For sending traffic, we will use both wrk and curl.
+  - *wrk* will generate a lot of traffic continuously, and it will have a high chance of catching of any
+    (however small) periods of downtime.
+  - *curl* will generate 1 request every 0.1s. While it might not catch small periods of downtime, it will give us
+    timeline of failed request for big periods of downtime, which wrk doesn't do.
+
+## Tests
+
+Run the following tests on both the 10 and 25 node clusters.
+
+### Setup
+
+1. Create a cluster.
+2. Deploy a previous latest stable version with 1 replica
+
+   For the 25 node cluster:
+
+    ```console
+     helm install my-release oci://ghcr.io/nginxinc/charts/nginx-gateway-fabric  --create-namespace --wait -n nginx-gateway --version 0.0.0-edge --values ./values-25-node.yaml
+    ```
+
+   For the 10 node cluster:
+
+    ```console
+     helm install my-release oci://ghcr.io/nginxinc/charts/nginx-gateway-fabric  --create-namespace --wait -n nginx-gateway --version 0.0.0-edge --values ./values-10-node.yaml
+    ```
+
+3. Expose NGF via a Service Load Balancer, internal (only accessible within the Google Cloud region) by adding
+   `networking.gke.io/load-balancer-type: "Internal"` annotation to the Service.
```

Comment:
This is done via the previous helm install command, right?

Reply:
yep, thanks for catching. I'll remove